src/charm.py

```diff
@@ -254,6 +254,9 @@ def _update(self, event):  # noqa: C901
             return
 
         context = {}
+        context.update(
+            {convert_env_var(key): value for key, value in self.config.items() if key not in ["environment"]}
```

Comment:
question: Why not `key != "environment"`? Did you meant to do something else here?

E: I see that this line just changed places. Still, if you (or the author) can tell why it is written this way that would be nice.

Reply:
All config variables on the charm are being converted into environment variables to be injected into the workload container, with the exception of the `environment` config which is a yaml-based config for configuring secrets introduced in #22 .